src/cljs/harja/tiedot/urakka/paikkaukset_toteumat.cljs

```diff
@@ -0,0 +1,143 @@
+(ns harja.tiedot.urakka.paikkaukset-toteumat
+  (:require [reagent.core :refer [atom] :as r]
+            [tuck.core :as tuck]
+            [harja.pvm :as pvm]
+            [harja.tiedot.navigaatio :as nav]
+            [harja.tyokalut.tuck :as tt]
+            [harja.ui.viesti :as viesti]
+            [harja.ui.grid :as grid]
+            [harja.domain.paikkaus :as paikkaus]
+            [harja.domain.tierekisteri :as tierekisteri])
+  (:require-macros [cljs.core.async.macros :refer [go]]))
+
+(def app (atom {:paikkauksien-haku-kaynnissa? false
+                :valinnat nil}))
+
+(defn kiinnostavat-tiedot-grid [{tierekisteriosoite ::paikkaus/tierekisteriosoite paikkauskohde ::paikkaus/paikkauskohde
+                                 :as paikkaus}]
+  (let [sellaisenaan-naytettavat-arvot (select-keys paikkaus #{::paikkaus/tyomenetelma
+                                                               ::paikkaus/massatyyppi ::paikkaus/leveys ::paikkaus/massamenekki
+                                                               ::paikkaus/raekoko ::paikkaus/kuulamylly ::paikkaus/id})
+        alkuaika {::paikkaus/alkuaika (pvm/pvm (::paikkaus/alkuaika paikkaus))}
+        loppuaika {::paikkaus/loppuaika (pvm/pvm (::paikkaus/loppuaika paikkaus))}
+        nimi (select-keys paikkauskohde #{::paikkaus/nimi})]
+    (merge sellaisenaan-naytettavat-arvot alkuaika loppuaika tierekisteriosoite nimi)))
+
+(defn kasittele-haettu-tulos
+  [tulos]
+  (let [kiinnostavat-tiedot (map #(kiinnostavat-tiedot-grid %)
+                                 tulos)
+        paikkaukset-grid (mapcat (fn [[otsikko paikkaukset]]
+                                   (cons (grid/otsikko otsikko) paikkaukset))
+                                 (group-by ::paikkaus/nimi kiinnostavat-tiedot))
+        paikkauket-vetolaatikko (map #(select-keys % [::paikkaus/tienkohdat ::paikkaus/materiaalit ::paikkaus/id])
+                                     tulos)]
+    {:paikkaukset-grid paikkaukset-grid
+     :paikkauket-vetolaatikko paikkauket-vetolaatikko}))
```

Comment:
Menee hiusten halkomiseksi, mutta hyvä separatio of concerns -periaate olisi, että tiedot namespace olisi täysin tiedoton näkymästä. Eli tässä ei terminologiaan olisi kiinni asioissa kuten "grid" ja "vetolaatikko".